scripts/scrapper.ts

```diff
@@ -1,10 +1,44 @@
 import axios from 'axios'
 import * as cheerio from 'cheerio'
-import { MealCategory } from '#models/meal'
+import Meal, { MealCategory } from '#models/meal'
+import { createHash } from 'node:crypto'
+import db from '@adonisjs/lucid/services/db'
+import WebsiteHash from '#models/website_hash'
+import logger from '@adonisjs/core/services/logger'
 
-export async function scrapeMenu() {
-  const url = 'https://sks.pwr.edu.pl/menu/'
+const url = 'https://sks.pwr.edu.pl/menu/'
+
+export async function runScrapper() {
+  const currentHash = await cacheMenu()
+  const storedHash = await WebsiteHash.first()
+
+  const trx = await db.transaction()
+  try {
+    if (!storedHash) {
+      logger.info('No hash found in database. Storing current hash and scraping menu.')
+      await WebsiteHash.create({ hash: currentHash }, { client: trx })
+    } else if (storedHash.hash === currentHash) {
+      logger.info('Did not find any differences. Not proceeding with scraping.')
+      return
```

Comment:
so I have no idea how lucid works, but I would check or ask Linek if there isn't any memory leak in this situation, when you create a trasaction, but you don't commit nor rollback it

Reply:
@dawidlinek what do u think about this?